Build failure: gpodder-3.11.4

### Steps To Reproduce

Steps to reproduce the behavior:

with gpodder as a systemPackage, I let the (daily) run of the nixos-upgrade.service do its thing

### Build log

```
...
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]: error: builder for '/nix/store/a1m9zs5ksr9wrasgic54bvsh9yi0352m-gpodder-3.11.4.drv' failed with exit code 2;
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        last 10 log lines:
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > msgfmt po/sv.po -o share/locale/sv/LC_MESSAGES/gpodder.mo
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > msgfmt po/tr.po -o share/locale/tr/LC_MESSAGES/gpodder.mo
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > msgfmt po/uk.po -o share/locale/uk/LC_MESSAGES/gpodder.mo
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > msgfmt po/zh_CN.po -o share/locale/zh_CN/LC_MESSAGES/gpodder.mo
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > python3 setup.py install --root=/ --prefix=/nix/store/j7kxjycpqgkb5kipvy1gzdk88am218cx-gpodder-3.11.4 --optimize=1
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > Traceback (most recent call last):
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        >   File "/build/source/setup.py", line 24, in <module>
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        >     from distutils.core import setup
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > ModuleNotFoundError: No module named 'distutils'
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        > make: *** [makefile:97: install] Error 1
Jul 20 06:23:36 nixos nixos-upgrade-start[3865]:        For full logs, run 'nix log /nix/store/a1m9zs5ksr9wrasgic54bvsh9yi0352m-gpodder-3.11.4.drv'.
```

### Additional context

there seems to be several related issues, something to do with the removal of distutils from Python 3.12?

### Notify maintainers

@mic92

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.11 (Vicuna), 24.11pre647193.9f4128e00b0a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(root): `"nixos"`
 - channels(oo): `"nixpkgs"`
 - nixpkgs: `/home/oo/.nix-defexpr/channels/nixpkgs`
```

---

--- a/pkgs/applications/audio/gpodder/default.nix
+++ b/pkgs/applications/audio/gpodder/default.nix
@@ -6,13 +6,12 @@
 , gobject-introspection
 , gtk3
 , intltool
-, python3
-, python3Packages
+, python311Packages
 , wrapGAppsHook3
 , xdg-utils
 }:
 
-python3Packages.buildPythonApplication rec {
+python311Packages.buildPythonApplication rec {
   pname = "gpodder";
   version = "3.11.4";
   format = "other";
@@ -40,12 +39,11 @@ python3Packages.buildPythonApplication rec {
   ];
 
   buildInputs = [
-    python3
     gtk3
     adwaita-icon-theme
   ];
 
-  nativeCheckInputs = with python3Packages; [
+  nativeCheckInputs = with python311Packages; [
     minimock
     pytest
     pytest-httpserver
@@ -54,7 +52,7 @@ python3Packages.buildPythonApplication rec {
 
   doCheck = true;
 
-  propagatedBuildInputs = with python3Packages; [
+  propagatedBuildInputs = with python311Packages; [
     feedparser
     dbus-python
     mygpoclient

